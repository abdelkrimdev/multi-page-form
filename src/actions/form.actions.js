import { formConstants } from '../constants/form.constants'

export const formActions = {
  startRegitration: () => {
    return { type: formConstants.START_REGISTRATION }
  },
  nextStep: () => {
    return { type: formConstants.NEXT_STEP }
  },
  prevStep: () => {
    return { type: formConstants.PREV_STEP }
  },
  updateForm: (input) => {
    return {
      type: formConstants.UPDATE_FORM,
      payload: input
    }
  }
}
