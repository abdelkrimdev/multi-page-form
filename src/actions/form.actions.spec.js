import configureStore from 'redux-mock-store'

import { formActions } from './form.actions'

const mockStore = configureStore()

describe('form actions', () => {
  it('should create start registration action.', () => {
    const store = mockStore()

    store.dispatch(formActions.startRegitration())

    expect(store.getActions()).toMatchSnapshot()
  })
  
  it('should create next step action.', () => {
    const store = mockStore()

    store.dispatch(formActions.nextStep())

    expect(store.getActions()).toMatchSnapshot()
  })

  it('should create previous step action.', () => {
    const store = mockStore()

    store.dispatch(formActions.prevStep())

    expect(store.getActions()).toMatchSnapshot()
  })

  it('should create update form action.', () => {
    const store = mockStore()
    const input = {
        key: 'name',
        value: 'Bruce'
    }

    store.dispatch(formActions.updateForm(input))

    expect(store.getActions()).toMatchSnapshot()
  })
})
