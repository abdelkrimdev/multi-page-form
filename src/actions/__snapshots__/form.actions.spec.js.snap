// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`form actions should create next step action. 1`] = `
Array [
  Object {
    "type": "GOTO_NEXT_STEP",
  },
]
`;

exports[`form actions should create previous step action. 1`] = `
Array [
  Object {
    "type": "GOTO_PREVIOUS_STEP",
  },
]
`;

exports[`form actions should create start registration action. 1`] = `
Array [
  Object {
    "type": "INITIALIZE_FORM_REGISTRATION",
  },
]
`;

exports[`form actions should create update form action. 1`] = `
Array [
  Object {
    "payload": Object {
      "key": "name",
      "value": "Bruce",
    },
    "type": "UPDATE_FORM_DATA",
  },
]
`;
